src/main.js

```diff
@@ -323,6 +344,13 @@ module.exports = {
       await waitOnIdle()
     }
 
+    // Sometimes the worker dies and becomes disconnected
```

Comment:
Since this is duplicated maybe this comment as well as the check should be moved into the called function, and it renamed to something like `checkESLintWorker`?

Reply:
I went back and forth on it, and in the end decided against passing yet another argument to sendJob, but can do so if you'd prefer that.

Reply:
I actually rather like the idea of just sticking this logic in `sendJob`, that way the logic for maintaining the worker stays there leaving the callers to just worry about how the job is used.

I'm not sure why moving the logic there would require an additional argument though?

Reply:
Is there a way to kill and restart a worker simply from a reference to that worker?  You can see that my `restartESLintWorker()` method sets `this.worker`, which is then passed to `sendJob`.  So I assume we would need to send a reference to `this.restartESLintWorker` to `sendJob` as well.  Or am I missing another approach that you have in mind?

Reply:
I was thinking this bit:
```js
if (this.worker && !this.worker.childProcess.connected) {
  await this.initializeWorker()
}
```

As well as the code in `initializeWorker` could be moved into `sendJob` (or called from there).

Reply:
Hm, I guess I'm not seeing a clear vision of what you have in mind.  Do you want to take a crack at a commit on this PR?

Reply:
I'm not sure if this is at all an improvement. A lot of side effects and misdirection, but I'll share the idea anyways.

````
const advancedWorker = worker => ({
  initialize: () => {
    this.initializeEslintWorker();
    return this.worker
  },
  restart: {
    this.restartEslintWorker,
    return this.worker
  }
  task: this.worker
})

function sendJob( worker, config) {
  if (!worker.task || !this.worker.childProcess.connected) {
    worker.task = worker.initialize()
  }
  // ....
}
```